src/plugins/workspace/public/components/workspace_creator/workspace_creator.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback } from 'react';
+import { EuiPage, EuiPageBody, EuiPageHeader, EuiPageContent, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { WorkspaceForm, WorkspaceFormSubmitData } from './workspace_form';
+import { WORKSPACE_OVERVIEW_APP_ID, WORKSPACE_OP_TYPE_CREATE } from '../../../common/constants';
+import { formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
+import { WorkspaceClient } from '../../workspace_client';
+
+export const WorkspaceCreator = () => {
+  const {
+    services: { application, notifications, http, workspaceClient },
+  } = useOpenSearchDashboards<{ workspaceClient: WorkspaceClient }>();
+
+  const isPermissionEnabled = application?.capabilities.workspaces.permissionEnabled;
+  const handleWorkspaceFormSubmit = useCallback(
+    async (data: WorkspaceFormSubmitData) => {
+      let result;
+      try {
+        const { permissions, ...attributes } = data;
+        result = await workspaceClient.create(attributes, permissions);
+      } catch (error) {
+        notifications?.toasts.addDanger({
+          title: i18n.translate('workspace.create.failed', {
+            defaultMessage: 'Failed to create workspace',
+          }),
+          text: error instanceof Error ? error.message : JSON.stringify(error),
+        });
+        return;
+      }
+      if (result?.success) {
+        notifications?.toasts.addSuccess({
+          title: i18n.translate('workspace.create.success', {
+            defaultMessage: 'Create workspace successfully',
+          }),
+        });
+        if (application && http) {
+          window.location.href = formatUrlWithWorkspaceId(
```

Comment:
Do we need to set a timeout before jumping to the workspace?

Reply:
Is the workspace id stored in the hash part? If this part of refactor was been finalized, we can use `application.navigateToUrl` instead. Then the page won't be reload. The toasts will be kept. Then we don't need to add delay.

Reply:
I see, we will use the path option(current behavior in workspace branch) to store the workspace id. So I guess we may need a timeout here.

Reply:
Will add an 1s timeout here.